### PR TITLE
refactor(scripts): modularize script files

### PR DESCRIPTION
**Pull request by glevco**
### Motivation

Refactor code so the single `scripts.py` file is broken into multiple modules, improving organization and maintainability. No behavior is changed by this PR, code is only moved between files. The only active changes are in imports to fix circular dependencies after the refactor.

### Acceptance Criteria

1. Move HathorScript to its own file 
1. Move BaseScript to its own file 
1. Move P2PKH to its own file 
1. Move MultiSig to its own file 
1. Move NanoContractMatchValues to its own file 
1. Move Opcode and functions to their own file
1. Remaining functions from the `scripts` file moved to either `construct` or `execute` modules
1. Fix imports and linter issues 

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

File: hathor/transaction/script/scripts.py
Comment (msbrogli) [inline hathor/transaction/script/scripts.py:1]: It seems this file still mixes things from "script execution" to "script utils to parse and build scripts".

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: hathor/transaction/script/scripts.py
Comment (glevco) [reply hathor/transaction/script/scripts.py:1]: Done! In two separate commits: https://github.com/HathorNetwork/hathor-core/pull/811/commits/6919f49bfaac10a147c0a1d5bd8e1df05628af84 and https://github.com/HathorNetwork/hathor-core/pull/811/commits/15050e3aaaffeea2957bb1651e3dea08ff21adfa

Comment (msbrogli) [reply hathor/transaction/script/scripts.py:1]: Thanks!

File: hathor/transaction/script/opcode.py [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+    EqualVerifyFailed,
+    InvalidScriptError,
+    InvalidStackData,
+    MissingStackItems,
+    OracleChecksigFailed,
+    ScriptError,
+    TimeLocked,
+    VerifyFailed,
+)
+from hathor.transaction.script.construct import (
```
Comment (msbrogli) [inline hathor/transaction/script/opcode.py:41]: I feel all these methods should be in the `hathor.transaction.script.execution` module. Even though they are also used by parsers and constructors, they were developed to be used during script execution.

File: hathor/transaction/script/__init__.py
Comment (msbrogli) [inline hathor/transaction/script/__init__.py:1]: Why not import common methods and classes here so we don't have to change so many imports all over the project?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: hathor/transaction/script/opcode.py (hunk 1 above)
Comment (glevco) [reply hathor/transaction/script/opcode.py]: Done!

File: hathor/transaction/script/__init__.py
Comment (glevco) [reply hathor/transaction/script/__init__.py:1]: Good idea, done! I ended up rebasing and pushing a single commit, as now the number of changed files was dramatically reduced.

**Review by msbrogli — APPROVED** [review]

File: hathor/transaction/script/opcode.py (hunk 1 above)
Comment (msbrogli) [reply hathor/transaction/script/opcode.py]: Thanks!

File: hathor/transaction/script/__init__.py
Comment (msbrogli) [reply hathor/transaction/script/__init__.py:1]: Well done! Thanks!

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

**Pull request merged**